lib/clone.js

```diff
@@ -0,0 +1,66 @@
+export const getFinalError = (initialError, message) => {
+	const error = createFinalError(initialError, message);
+	previousErrors.add(error);
+	return error;
+};
+
+const createFinalError = (error, message) => {
+	if (Object.prototype.toString.call(error) !== '[object Error]') {
+		return new Error(message);
+	}
+
+	return previousErrors.has(error)
+		? cloneError(error, message)
+		: setErrorMessage(error, message);
+};
+
+const cloneError = (oldError, newMessage) => {
+	const {name, message, stack} = oldError;
+	const error = new Error(newMessage);
+	error.stack = fixStack(name, stack, message, newMessage);
+	Object.defineProperty(error, 'name', {value: name, enumerable: false, configurable: true, writable: true});
```

Comment:
Unlike the other properties below, `error.name` is usually set on the prototype.